### PR TITLE
Temporarily change tests to accommodate new `SemanticsAction` value

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
This is for fixing a test failure in an engine PR that introduces a new `SemanticsAction`: https://github.com/flutter/engine/pull/56606/files

The test will be updated in the future to remove the index-based predicate.

failure: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8730901009848520305/+/u/run_test.dart_for_web_tests_shard_and_subshard_1/stdout

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/test/widgets/semantics_test.dart [hunk 1]
```diff
@@ -524,7 +524,8 @@ void main() {
       ..remove(SemanticsAction.moveCursorForwardByWord)
       ..remove(SemanticsAction.moveCursorBackwardByWord)
       ..remove(SemanticsAction.customAction) // customAction is not user-exposed.
-      ..remove(SemanticsAction.showOnScreen); // showOnScreen is not user-exposed
+      ..remove(SemanticsAction.showOnScreen) // showOnScreen is not user-exposed
+      ..removeWhere((SemanticsAction action) => action.index == 1 << 23);
```
Comment (chunhtai) [inline packages/flutter/test/widgets/semantics_test.dart:528]: Can you add a todo so that we don't forget this?

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/test/widgets/semantics_test.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/semantics_test.dart:528]: I'll only merge this if the other PR is ready so I can immediately update the test, sounds good?

Comment (chunhtai) [reply packages/flutter/test/widgets/semantics_test.dart:528]: ok, sgtm

Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/semantics_test.dart:528]: @chunhtai do you think can merge both PRs now?

Comment (chunhtai) [reply packages/flutter/test/widgets/semantics_test.dart:528]: let me take a quick look at the engine pr now

Comment (chunhtai) [reply packages/flutter/test/widgets/semantics_test.dart:528]: lgtm'd

**Pull request merged**